### ISSUE TITLE
error in SDAECriterionGPU.lua

### ISSUE THREAD

**Issue by caugusta** [fstrub95/Autoencoders_cf#7]
Running the code from the summary:
git clone git@github.com:fstrub95/Autoencoders_cf.git
cd Autoencoders_cf
cd data
wget http://files.grouplens.org/datasets/movielens/ml-10m.zip
unzip ml-10m.zip 
cd ../src
th data.lua  -ratings ../data/ml-10M100K/ratings.dat -metaItem ../data/ml-10M100K/movies.dat -out ../data/ml-10M100K/movieLens-10M.t7 -fileType movieLens -ratio 0.9
th main.lua  -file ../data/ml-10M100K/movieLens-10M.t7 -conf ../conf/conf.movieLens.10M.V.lua  -save network.t7 -type V -meta 1 -gpu 1

I get stuck on the last line (main.lua):

~/Autoencoders_cf/src$ th main.lua  -file ../data/ml-10M100K/movieLens-10M.t7 -conf ../conf/conf.movieLens.10M.V.lua  -save network.t7 -type V -meta 1 -gpu 1
Options:    
- gpu       : 1  
- file      : ../data/ml-10M100K/movieLens-10M.t7  
- seed      : 0  
- save      : network.t7  
- conf      : ../conf/conf.movieLens.10M.V.lua 
- type      : V  
- meta      : 1  
  Load training configuration...  
  Load data...  
  Loading data from disk...  

Users loaded      : 69878  
Items loaded      : 10677   

No Train ratings  : 9001572 
No Test  ratings  : 998482  
Training ratio    : 0.90015233917737  
Training density  : 0.012065036787392   

Loading cunn... 
Loading data to GPU...  
Unbias the data...  
Data was successfully preloaded...  
Training network  
Start training : layer1-1  
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> output](1): cfn.AppenderSparseOut
  (2): nnsparse.Densify
  (3): nn.Linear(69897 -> 770)
  (4): nn.Tanh
  (5): cfn.AppenderOut
  (6): nn.Linear(789 -> 69878)
  (7): nn.Tanh
}
 [>............................... 1/50 .......................................] ETA: 0ms | Step: 0ms  
**/usr/local/bin/luajit: tools/SDAECriterionGPU.lua:65: invalid arguments: CudaTensor CudaTensor CudaTensor 
expected arguments: [_CudaTensor_] CudaTensor CudaByteTensor**
stack traceback:
    [C]: in function 'maskedSelect'
    tools/SDAECriterionGPU.lua:65: in function 'prepareInput'
    misc/AutoEncoderTrainer.lua:97: in function 'opfunc'
    /usr/local/share/lua/5.1/optim/sgd.lua:43: in function 'sgd'
    misc/AutoEncoderTrainer.lua:121: in function 'Train'
    misc/AutoEncoderTrainer.lua:266: in function 'Execute'
    misc/TrainNetwork.lua:150: in function 'TrainNetwork'
    main.lua:69: in main chunk
    [C]: in function 'dofile'
    /usr/local/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
    [C]: at 0x00406260

**Comment by fstrub95** [comment]
I reproduce the error. I implemented the fix and I am running a bunch of experience to be sure that there is no regression. I should upload the fix by the end of the week

**Comment by leetiantian** [comment]
I have the same problem

**Comment by fstrub95** [comment]
Fixed! At least!
I move the sparse operation into the CPU. I first implemented full GPU operation but it was far too slow.
Sorry for the delay! If you face some problem, tell me
